prow/cmd/pipeline/controller.go

```diff
@@ -416,9 +502,16 @@ func reconcile(c reconciler, key string) error {
 	case finalState(pj.Status.State):
 		logrus.Infof("Observed finished: %s", key)
 		return nil
+	case cancelledState(pj.Status.State):
+		if p != nil && p.Spec.Status != v1beta1.PipelineRunSpecStatusCancelled {
+			if _, err = c.cancelPipelineRun(ctx, p); err != nil {
```

Comment:
So why does `cancel` return an instance when the only caller discards it?

Reply:
The function underneath (patch tekton something...) returns it, but it could be removed as well. What do you prefer?

Reply:
I'd prefer to remove it if it isn't used